Assets/Scripts/MonoBehaviours/WeaponsSystem/WeaponBehaviourScript.cs

```diff
@@ -0,0 +1,148 @@
+        private Rigidbody2D _rigidbody;
+        private Vector2 _direction;
+    
+        /// <summary>
+        /// Returns the world space position of where the weapon's muzzle is
+        /// </summary>
+        public Vector2 WeaponSpriteEndPosition => (Vector2) transform.position + (_direction * new Vector2(_spriteRenderer.sprite.bounds.extents.x, _spriteRenderer.sprite.bounds.extents.y));
+        public WeaponData WeaponData => weaponData;
+
+        /// <summary>
+        /// Property for the weapon's direction
+        /// </summary>
+        public Vector2 Direction
+        {
+            get => _direction;
+            set
+            {
+                _direction = value;
+                if (_direction == Vector2.left)
+                {
+                    _spriteRenderer.flipX = true;
+                } else if (_direction.Equals(Vector2.right))
+                {
+                    _spriteRenderer.flipX = false;
+                }
+            }
+        }
+
+        public WeaponState WeaponStateProp
+        {
+            get => weaponState;
+            set => weaponState = value;
+        }
+
+        public int CurrentMagazineAmmunition
+        {
+            get => currentMagazineAmmunition;
+            set => currentMagazineAmmunition = value;
+        }
+
+        public int CurrentTotalAmmunition
+        {
+            get => currentTotalAmmunition;
+            set => currentTotalAmmunition = value;
+        }
+
+        private void Awake()
+        {
+            _rigidbody = GetComponent<Rigidbody2D>();
+            _rigidbody.sleepMode = RigidbodySleepMode2D.StartAsleep;
+            _spriteRenderer = GetComponent<SpriteRenderer>();
+            _weaponOnGroundBehaviour = GetComponent<WeaponOnGroundBehaviour>();
+            _weaponOnGroundBehaviour.Init(weaponData);
+        
+            //TODO remove this
+            Direction = Vector2.left;
+        }
+
+        private void Update()
+        {
+            switch (weaponState)
+            {
+                case WeaponState.OnGround:
+                    _weaponOnGroundBehaviour.enabled = true;
+                    _rigidbody.WakeUp();
+                    break;
+                case WeaponState.InInventory:
+                    _weaponOnGroundBehaviour.enabled = false;
+                    _spriteRenderer.sprite = weaponData.InInventorySprite;
+                    _rigidbody.Sleep();
+                    break;
+                case WeaponState.Inactive:
+                    _weaponOnGroundBehaviour.enabled = false;
+                    _spriteRenderer.sprite = weaponData.InactiveSprite;
+                    break;
+                case WeaponState.Active:
+                    _weaponOnGroundBehaviour.enabled = false;
+                    _spriteRenderer.sprite = weaponData.ActiveSprite;
+                    break;
+                default:
+                    throw new ArgumentOutOfRangeException();
+            }
```

Comment:
This could be abstracted into its own method because the Update might get quite large and complex over the course of the project

Reply:
I'd say that right now it's fine the way it is. If it gets any longer then we can create a new method